tests/fault_tolerance/deploy/scenarios.py

```diff
@@ -569,12 +563,22 @@ def create_legacy_load(
         # Get model from deployment info or use the global model
         scenario_model = deployment_info.get("model", model)
 
+        # Determine custom validation if needed
+        # For now, let pattern matching handle it, but you can override here
+        custom_validation = None
+
+        # Example: Add custom validation for specific scenario
+        # if scenario_name == "vllm-agg-tp-1-dp-1-decode_worker_pod":
+        #     from tests.fault_tolerance.deploy.validations import validate_decode_worker_pod_deletion
+        #     custom_validation = validate_decode_worker_pod_deletion
+
         scenarios[scenario_name] = Scenario(
             deployment=deployment_info["spec"],
             load=load_config,
             failures=failure,
             model=scenario_model,
             backend=backend,
+            validation=custom_validation,
```

Comment:
I almost think it would be clearer to have this added explicitly - all checks added as a list to the scenario -